inspirehep/modules/workflows/tasks/actions.py

```diff
@@ -204,3 +224,24 @@ def _prepare_update_payload(obj, eng):
         # FIXME: Just update entire record for now
         obj.extra_data[extra_data_key] = obj.data
     return _prepare_update_payload
+
+
+@with_debug_logging
+def refextract(obj, eng):
+    """Extract references from PDF.
+
+    :param obj: Workflow Object to process
+    :param eng: Workflow Engine processing the object
+    """
+    uri = get_pdf_in_workflow(obj)
+    if uri:
+        mapped_references = extract_references(uri)
+        if mapped_references:
+            obj.data["references"] = mapped_references
+            obj.log.info("Extracted {0} references".format(
+                len(mapped_references)
+            ))
+        else:
+            obj.log.info("No references extracted")
+    else:
+        obj.log.error("Not able to download and process the PDF")
```

Comment:
Uhm, I guess that we had already some repetition in the code: https://github.com/inspirehep/inspire-next/blob/f173f92b7db9294d3c25de3690634a82c2d39ab2/inspirehep/modules/workflows/tasks/refextract.py#L87-L134